### DISCUSSION TITLE
Metric for Network partition detected

### DISCUSSION THREAD

**Discussion by Sallaxer** [rabbitmq/rabbitmq-server#10558] [Questions] [answered]
### Is your feature request related to a problem? Please describe.

Cluster status - cluster has experienced a network partition

### Describe the solution you'd like

Metric that shows the status of the cluster: **Network partition detected**

### Describe alternatives you've considered

A third party exporter has a metric `rabbitmq_partitions` that can be used to set up alerts for this cluster state

### Additional context

I am using RabbitMQ version **3.12.12** and Erlang **26.2.1**. I see that there is a metric `rabbitmq_unreachable_cluster_peers_count`. But its value is `0` while in the web interface: ***Network partition detected
Mnesia reports that this RabbitMQ cluster has experienced a network partition. There is a risk of losing data***
This behavior can be easily reproduced, for example, if you block on one of the nodes the other nodes in the cluster (using iptables) and then unblock them again. Metrics in Prometheus will be available from all nodes, but if a third-party exporter shows a problem using the `rabbitmq_partitions` metric, the built-in one does not transmit such metrics and how to know that the state of the cluster is out of order is not clear (what kind of alert can be configured?).

**Comment by lukebakken** [comment, marked as answer]
Hello, and thanks for using RabbitMQ.

`rabbitmq_unreachable_cluster_peers_count` was introduced here:

* https://github.com/rabbitmq/rabbitmq-server/issues/2508
* https://github.com/rabbitmq/rabbitmq-server/pull/9465

If there is an issue that can be reproduced, please provide _EXACT_ steps for doing so.

> if you block on one of the nodes the other nodes in the cluster (using iptables) and then unblock them again

This might _seem_ easy but anything you can do (like provide the exact commands you ran) would help us focus on the issue rather than wasting time trying to reproduce it. Even better would be a docker compose project demonstrating the issue.

If we find an actual bug, an issue can be filed.

**Comment by michaelklishin** [comment]
The idea behind the above metric was exactly the same as what you have in mind.

However, this metric will work fine in the world of 4.x/Khepri where most “partition handling” states will go away entirely since the recovery algorithm for everything will be that of Raft.

Therefore introducing a different version of these metrics does not make much sense.

This metric also matches the Prometheus take on metric scraping while cluster state metrics generally do not.

**Reply by Sallaxer** [reply]
Thanks for the quick reply )
Ok, am I understanding correctly that right now it is not possible to track this cluster state by the `rabbitmq_unreachable_cluster_peers_count` metric or indirectly any other metric? In future 4.x it will respond correctly to this state? And of the options, for now only using a third-party exporter?
